src/PrestaShopBundle/Bridge/Controller/ControllerConfiguration.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Controller;
+
+use PrestaShopBundle\Security\Admin\Employee;
+
+/**
+ * Define configuration for Controller migrate horizontally
+ */
+class ControllerConfiguration
+{
+    /**
+     * @var int
+     */
+    public $id;
```

Comment:
Public properties are really a good idea here ? There is no way to restrict property types, to restrict access outside of this class.
If this configuration was a simple DTO only, public properties would be fine. But this is the corner stone of horizontal migration, and it might require strict access control.
OO design usually rely on methods to override, not on object properties.
Here it means a dedicated interface might be useful to design the contract with other components, to implement custom configuration as well if necessary.

Reply:
Or simply to override the standard behaviour whenever required. The power of OO design lays in methods, not in properties.